src/libexpr/eval-error.cc

```diff
@@ -0,0 +1,99 @@
+#include "eval-error.hh"
```

Comment:
question about why not just write the implementation in header file, maybe `eval-error.hh`?

Reply:
Reduces compile times, at least to an extent. I can put the implementation in the header file if that's what's preferred, though.

Reply:
> Reduces compile times, at least to an extent

If you only have implememtations in .cc files(formally translation unit, TUs), other libexpr users may COPY the source code again and do same instantiation.

> Reduces compile times

I think it is not a good option to do any 'optimization' without dedicated profiling. This *may* reduce compile time, but *do* make things hard somehow(as I commented above).

Reply:
I don't mind this so much. If it becomes annoying it's easy to change.

Reply:
I went to refactor this and remembered why I had it like this -- the `PosIdx` type is forward-declared (it comes from `nixexpr.hh`, which itself depends on `eval-error.hh`).

Reply:
Unfortunately even if you move `PosIdx` out of `nixexpr.hh`, the definition of `EvalErrorBuilder<T>::withFrame` depends on `EvalState` :(

Reply:
> Unfortunately even if you move `PosIdx` out of `nixexpr.hh`, the definition of `EvalErrorBuilder<T>::withFrame` depends on `EvalState` :(

That's quite unfortunate. Ideally all headers are 'self-contained', looks like those many 'forward declaration's lead to such isssue. This cannot be trivially solved, then let's keep it as-is :(.